invoke.go

```diff
@@ -11,10 +12,13 @@ import (
 // A user defined call stub.
 type APICall func(context.Context) error
 
-// scaleDuration returns the product of a and mult.
-func scaleDuration(a time.Duration, mult float64) time.Duration {
-	ns := float64(a) * mult
-	return time.Duration(ns)
+// scaleDuration returns the duration in `mult` after `d`
```

Comment:
This change was made so that we don't need to create 3 new contexts every time we want to sleep at https://github.com/googleapis/gax-go/pull/35/files#diff-39e0330b64fba4170680bf6bd806b22fL51 .

Since I need to rewrite a lot of invokeWithRetry anyway (this function isn't used elsewhere), I thought I might as well simplify it. Would you like me to separate this into another change?